client/consensus/grandpa/aux_schema.go

```diff
@@ -0,0 +1,238 @@
+
+		state := finalityGrandpa.NewRoundState(finalityGrandpa.HashNumber[H, N]{Hash: genesisHash, Number: genesisNumber})
+		base := state.PrevoteGHOST
+		if base == nil {
+			panic("state is for completed round; completed rounds must have a prevote ghost; qed.")
+		}
+
+		voterSetState := NewVoterSetState[H, N]()
+		genesisState, err := voterSetState.Live(0, *genesisSet, *base)
+		if err != nil {
+			return nil, err
+		}
+
+		insert := []KeyValue{
+			{AUTHORITY_SET_KEY, scale.MustMarshal(*genesisSet)},
+			{SET_STATE_KEY, scale.MustMarshal(genesisState)},
+		}
+
+		err = store.Insert(insert, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		return &persistentData[H, N]{
+			authoritySet: SharedAuthoritySet[H, N]{inner: *genesisSet},
+			setState:     SharedVoterSetState[H, N]{inner: genesisState},
+		}, nil
+	}
+}
+
+// UpdateAuthoritySet Update the authority set on disk after a change.
+//
+// If there has just been a handoff, pass a `new_set` parameter that describes the
+// handoff. `set` in all cases should reflect the current authority set, with all
+// changes and handoffs applied.
+func UpdateAuthoritySet[H comparable, N constraints.Unsigned](store AuxStore, set AuthoritySet[H, N], newSet *NewAuthoritySetStruct[H, N]) error {
+	// TODO make sure that Insert has affect of both insert and update depending on use case
+	encodedAuthSet, err := scale.Marshal(set)
+	if err != nil {
+		return err
+	}
+
+	if newSet != nil {
+		// we also overwrite the "last completed round" entry with a blank slate
+		// because from the perspective of the finality gadget, the chain has
+		// reset.
+		genesisState := finalityGrandpa.HashNumber[H, N]{
+			Hash:   newSet.CanonHash,
+			Number: newSet.CanonNumber,
+		}
+		voterSetState := NewVoterSetState[H, N]()
+		setState, err := voterSetState.Live(uint64(newSet.SetId), set, genesisState)
+		if err != nil {
+			return err
+		}
+
+		encodedVoterSet, err := scale.Marshal(setState)
+		if err != nil {
+			return err
+		}
+
+		insert := []KeyValue{
+			{AUTHORITY_SET_KEY, encodedAuthSet},
+			{SET_STATE_KEY, encodedVoterSet},
+		}
+		err = store.Insert(insert, nil)
+		if err != nil {
+			return err
+		}
+
+	} else {
+		insert := []KeyValue{
+			{AUTHORITY_SET_KEY, encodedAuthSet},
+		}
+
+		err = store.Insert(insert, nil)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UpdateBestJustification Update the justification for the latest finalized block on-disk.
+//
+// We always keep around the justification for the best finalized block and overwrite it
+// as we finalize new blocks, this makes sure that we don't store useless justifications
+// but can always prove finality of the latest block.
+func UpdateBestJustification() {
+	// TODO impl when we have justification logic
+	panic("impl")
+}
+
+// BestJustification  Fetch the justification for the latest block finalized by GRANDPA, if any.
+func BestJustification() {
+	// TODO impl when we have justification logic
+	panic("impl")
+}
```

Comment:
Yes, because none of the justification logic is implemented and IMO it is enough work to separate into a different PR. This has already been done and addressed in #3454